sql/log_event.cc

```diff
@@ -2649,10 +2649,9 @@ Gtid_log_event::Gtid_log_event(const uchar *buf, uint event_len,
       extra engines flags presence is identifed by non-zero byte value
       at this point
     */
-    if (flags_extra & FL_EXTRA_MULTI_ENGINE)
+    if ((flags_extra & FL_EXTRA_MULTI_ENGINE) &&
+        (static_cast<uint>(buf - buf_0) < event_len))
```

Comment:
```
 if (flags2 & (FL_PREPARED_XA | FL_COMPLETED_XA))
  {
    xid.formatID= uint4korr(buf);
    buf+= 4;
```

You added a check for FL_EXTRA_MULTI_ENGINE, but why no similar check for the FL_PREPARED_XA / FL_COMPLETED_XA ?

Reply:
I only looked at `extra_flags` - nice spot. I'll add a check for those flags too.